client/src/main/java/org/devfleet/esi/api/service/ESIServiceFactory.java

```diff
@@ -0,0 +1,31 @@
+package org.devfleet.esi.api.service;
+
+import org.devfleet.esi.api.*;
+import org.devfleet.esi.api.Character;
+import org.devfleet.esi.api.service.ESICharacterService;
+import org.devfleet.esi.impl.CorporationData;
+import org.devfleet.esi.impl.MailboxData;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface ESIServiceFactory {
+
+    Optional<Character> tryFetchCharacter(CharacterKey key);
```

Comment:
I know Optional is popular but personally I never liked the idea. Returning null is a perfectly viable contract in most cases.